components/BarcodeScanner.vue

```diff
@@ -138,28 +155,51 @@ export default {
       cameraConstraints.zoom.step = step;
     }
 
+    async function emitScanResult({barcodeDetector, video}) {
+      const {signal} = abortController;
+      const barcodes = await detectBarcodes({barcodeDetector, video, signal});
+      const [result] = barcodes;
+      if(result?.format && result?.rawValue) {
+        emit('result', {type: result.format, text: result.rawValue});
+      }
```

Comment:
See comment above. If we need to handle an abort, we'll need to wrap `await detectBarcodes(...)` in a try/catch and do something when the catch occurs with the UI. But if the promise resolves (it doesn't reject/throw) then we know we have a result.

```suggestion
      emit('result', {type: result.format, text: result.rawValue});
```

Reply:
Addressed here 9497a7d